### PR TITLE
redo docker UI to catch up with middleware changes.

### PR DESCRIPTION
**Pull request by vaibhav-rbs**
this also fixes, #46025

### CODE REVIEW

**Comment by vaibhav-rbs** [comment]
please update your freenas machine /freenas vm with following as newly introduced middleware methods are in this PR: https://builds.ixsystems.com/jenkins/job/FreeNAS%20-%20PR/view/change-requests/job/PR-1795/lastSuccessfulBuild/artifact/artifacts/

File: src/app/pages/vm/dockervm-wizard/dockervm-wizard.component.ts [hunk 1]
```diff
@@ -315,41 +321,30 @@ async customSubmit(value) {
...
-        },(error) => {
-          this.loader.close();
-        });
-      }
-    },
-    (error_res) => {
-      new EntityUtils().handleError(this, error_res);
-      this.loader.close();
-    })
+    this.dialogRef = this.dialog.open(EntityJobComponent, { data: { "title": T("Docker VM") }});
```
Comment (lola27) [inline src/app/pages/vm/dockervm-wizard/dockervm-wizard.component.ts]: we should disable the `close` button in this situation.

File: src/app/pages/vm/dockervm-wizard/dockervm-wizard.component.ts [hunk 2]
```diff
@@ -315,41 +321,30 @@ async customSubmit(value) {
...
+    this.dialogRef.componentInstance.progressNumberType = "nopercent";
```
Comment (lola27) [inline src/app/pages/vm/dockervm-wizard/dockervm-wizard.component.ts]: 'nopercent' is not correct here.

**Comment by vaibhav-rbs** [comment]
rebuild

File: src/app/pages/vm/devices/device-add2/device-add.component.ts [hunk 3]
```diff
@@ -200,7 +200,7 @@ export class DeviceAddComponent implements OnInit {
       type : 'select',
       name : 'sectorsize',
       placeholder : 'Disk sector size',
-      tooltip : 'Enter a sector size in bytes. <i>0</i> leaves the\
+      tooltip : 'Select a sector size in bytes. <i>0</i> leaves the\
```
Comment (lola27) [inline src/app/pages/vm/devices/device-add2/device-add.component.ts]: '0 leaves the sector size unset', but the label for '0' option is Default. Could you make them consistent?

Comment (vaibhav-rbs) [reply src/app/pages/vm/devices/device-add2/device-add.component.ts]: ok will make changes right now

**Pull request merged**